core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -957,6 +957,17 @@ class LogManagerTest {
       val topicPartition = UnifiedLog.parseTopicPartitionName(dir)
       val config = topicConfigOverrides.getOrElse(topicPartition.topic, logConfig)
 
+      if (dir == logDir2) {
```

Comment:
Yep, this is the one of the solutions I described in the jira. It does not change production code but it can't fix the issue totally. 

If we don't want to touch production code, maybe another way is to change the folders from 2 to 1. Not sure whether this is valid to this test case.

Reply:
@FrankYang0529 Could you please try another approach? create the snapshot when holding the write lock?

Reply:
Yes, I tried another to get `epochs.values()` in the write lock, so we don't need to get read lock again to get the value.